src/sidebar/helpers/view-filter.js

```diff
@@ -21,52 +12,43 @@ import * as unicodeUtils from '../util/unicode';
  */
 
 /**
- * @typedef Checker
- * @prop {(ann: Annotation) => boolean} autofalse
- * @prop {(ann: Annotation) => any|any[]} value
- * @prop {(term: any, value: any) => boolean} match
+ * A Matcher specifies how to test whether an annotation matches a query term
+ * for a specific field.
+ *
+ * @template [T=string] - Type of parsed query terms and field values
+ * @typedef Matcher
+ * @prop {(ann: Annotation) => T[]} fieldValues - Extract the field values to be
+ *   matched against a query term
+ * @prop {(value: T, term: T) => boolean} matches - Test whether a query term
+ *   matches a field value. Both value and term will have been normalized using
+ *   `normalize`.
+ * @prop {(val: T) => T} normalize - Normalize a parsed term or field value for
+ *   comparison
  */
 
-/** @param {Annotation} ann */
-function displayName(ann) {
-  if (!ann.user_info) {
-    return '';
-  }
-  return ann.user_info.display_name || '';
-}
-
 /**
- * Normalize a field value or query term for comparison.
+ * Normalize a string query term or field value.
  *
- * @template {string|number} T
- * @param {T} val
+ * @param {string} val
  */
-function normalize(val) {
-  if (typeof val !== 'string') {
-    return val;
-  }
+function normalizeStr(val) {
```

Comment:
Completely outside the scope of this, but a curiosity: Do you have any notion of where the approach here came from? Why the decision was made to normalize a string in specifically these ways? Or just common sense?